### ISSUE TITLE
bug: unknown Webhook event "types" / types section is sequence node but mapping node is expected

### ISSUE THREAD

**Issue by cipherboy** [rhysd/actionlint#460]
When running against [this workflow](https://github.com/openbao/openbao/blob/101751d032c889dc86ca4bc737133b73012c5aa5/.github/workflows/verify-commits.yml), I get the error:

```
.github/workflows/verify-commits.yml:5:3: unknown Webhook event "types". see https://docs.github.com/en/actions/learn-github-actions/events-that-trigger-workflows#webhook-events for list of all Webhook event names [events]
  |
5 |   types: [opened, synchronize]
  |   ^~~~~~
.github/workflows/verify-commits.yml:5:10: "types" section is sequence node but mapping node is expected [syntax-check]
  |
5 |   types: [opened, synchronize]
  |          ^~~~~~~~
```

However, this section comes nearly verbatim [from the docs](https://docs.github.com/en/actions/writing-workflows/choosing-when-your-workflow-runs/events-that-trigger-workflows#pull_request):

```yaml
on:
  pull_request:
    types: [opened, reopened]
```

which also specify `synchronize` is a valid value. Additionally, it seems like this should be a sequence and not a mapping. 

If I read into this err correctly, it seems to suggest that `types` is being detected not as a property of `pull_request` but of `on` and its nesting level is not inferred correctly. This makes `actionlint` think it is an unknown Webhook event (called `types`) rather than a modifier of the `pull_request` webhook event.

**Comment by rhysd** [comment]
Please provide a reproducible example so that I can confirm the issue on my local machine.

**Comment by rhysd** [comment]
I copied your workflow into local and ran `actionlint`. It didn't report any error. Can you check `actionlint` is the latest? Here is the log of my trial:

```
> actionlint -version
1.7.3
installed by building from source
built with go1.23.1 compiler for darwin/amd64
> cat test.yaml
name: Ensure Verified Commits

on:
  pull_request:
    types: [opened, synchronize]

jobs:
  verify_commits:
    name: Verify Commit Signatures
    runs-on: ubuntu-latest

    steps:
      - uses: actions/checkout@v4
      - name: Check commit signatures
        run: |
          gh api \
            -H "Accept: application/vnd.github+json" \
            -H "X-GitHub-Api-Version: 2022-11-28" \
            /repos/openbao/openbao/pulls/${{ github.event.pull_request.number }}/commits | \
          jq -f .github/scripts/verify-commits.jq
        env:
          GH_TOKEN: ${{ github.token }}
> actionlint test.yaml
>
```

**Comment by cipherboy** [comment]
@rhysd Hmm, I'm happy if it isn't reproducible.

The run was: https://github.com/openbao/openbao-plugin-auth-jwt/actions/runs/11411272642/job/31755210861 
 
```
2024-10-18T21:27:38.9314696Z ##[group]Run docker://docker.mirror.hashicorp.services/rhysd/actionlint:latest
2024-10-18T21:27:38.9315454Z ##[endgroup]
2024-10-18T21:27:38.9518561Z ##[command]/usr/bin/docker run --name dockermirrorhashicorpservicesrhysdactionlintlatest_c4a406 --label 9a0311 --workdir /github/workspace --rm -e "HOME" -e "GITHUB_JOB" -e "GITHUB_REF" -e "GITHUB_SHA" -e "GITHUB_REPOSITORY" -e "GITHUB_REPOSITORY_OWNER" -e "GITHUB_REPOSITORY_OWNER_ID" -e "GITHUB_RUN_ID" -e "GITHUB_RUN_NUMBER" -e "GITHUB_RETENTION_DAYS" -e "GITHUB_RUN_ATTEMPT" -e "GITHUB_REPOSITORY_ID" -e "GITHUB_ACTOR_ID" -e "GITHUB_ACTOR" -e "GITHUB_TRIGGERING_ACTOR" -e "GITHUB_WORKFLOW" -e "GITHUB_HEAD_REF" -e "GITHUB_BASE_REF" -e "GITHUB_EVENT_NAME" -e "GITHUB_SERVER_URL" -e "GITHUB_API_URL" -e "GITHUB_GRAPHQL_URL" -e "GITHUB_REF_NAME" -e "GITHUB_REF_PROTECTED" -e "GITHUB_REF_TYPE" -e "GITHUB_WORKFLOW_REF" -e "GITHUB_WORKFLOW_SHA" -e "GITHUB_WORKSPACE" -e "GITHUB_ACTION" -e "GITHUB_EVENT_PATH" -e "GITHUB_ACTION_REPOSITORY" -e "GITHUB_ACTION_REF" -e "GITHUB_PATH" -e "GITHUB_ENV" -e "GITHUB_STEP_SUMMARY" -e "GITHUB_STATE" -e "GITHUB_OUTPUT" -e "RUNNER_OS" -e "RUNNER_ARCH" -e "RUNNER_NAME" -e "RUNNER_ENVIRONMENT" -e "RUNNER_TOOL_CACHE" -e "RUNNER_TEMP" -e "RUNNER_WORKSPACE" -e "ACTIONS_RUNTIME_URL" -e "ACTIONS_RUNTIME_TOKEN" -e "ACTIONS_CACHE_URL" -e "ACTIONS_RESULTS_URL" -e GITHUB_ACTIONS=true -e CI=true -v "/var/run/docker.sock":"/var/run/docker.sock" -v "/home/runner/work/_temp/_github_home":"/github/home" -v "/home/runner/work/_temp/_github_workflow":"/github/workflow" -v "/home/runner/work/_temp/_runner_file_commands":"/github/file_commands" -v "/home/runner/work/openbao-plugin-auth-jwt/openbao-plugin-auth-jwt":"/github/workspace" docker.mirror.hashicorp.services/rhysd/actionlint:latest
2024-10-18T21:27:39.2716671Z .github/workflows/verify-commits.yml:5:3: unknown Webhook event "types". see https://docs.github.com/en/actions/learn-github-actions/events-that-trigger-workflows#webhook-events for list of all Webhook event names [events]
2024-10-18T21:27:39.2719282Z   |
2024-10-18T21:27:39.2720246Z 5 |   types: [opened, synchronize]
2024-10-18T21:27:39.2720643Z   |   ^~~~~~
2024-10-18T21:27:39.2721464Z .github/workflows/verify-commits.yml:5:10: "types" section is sequence node but mapping node is expected [syntax-check]
2024-10-18T21:27:39.2722607Z   |
2024-10-18T21:27:39.2723089Z 5 |   types: [opened, synchronize]
2024-10-18T21:27:39.2724387Z   |          ^~~~~~~~
2024-10-18T21:27:39.4229864Z Post job cleanup.
2024-10-18T21:27:39.5169909Z [command]/usr/bin/git version
```

from [here](https://github.com/hashicorp/vault-workflows-common/blob/main/.github/workflows/actionlint.yaml) via [here](https://github.com/openbao/openbao-plugin-auth-jwt/blob/c82ad4a3684e1ef48c73d867342dcc8e1649e0a7/.github/workflows/actionlint.yaml).

I don't see a verbose image id there, but pulling locally I get the same value as on Dockerhub:

```
docker.io/rhysd/actionlint                                         latest          4c6870ee6f09  2 weeks ago    71.3 MB
docker.mirror.hashicorp.services/rhysd/actionlint                  latest          4c6870ee6f09  2 weeks ago    71.3 MB
```

And running the container locally seems fine:

```
$ podman run --mount type=bind,src=test.yaml,target=/test.yaml,chown=true,relabel=shared docker.mirror.hashicorp.services/rhysd/actionlint:latest --verbose /test.yaml
verbose: Linting test.yaml
verbose: Found 0 parse errors in 0 ms for test.yaml
verbose: Found total 0 errors in 5 ms for test.yaml
```

So I'm happy to call this a random failure in the Actions.

Thanks anyways and sorry for the noise!

**Comment by rhysd** [comment]
Thank you for the information. That sounds weird... I'll look at the log.
